### ISSUE TITLE
Interactive scrubber bar

### ISSUE THREAD

**Issue by BrechtDeMan** [BrechtDeMan/WebAudioEvaluationTool#238]
Some users evidently require functions such as 

- Moving backward (or forward) in a stimulus, especially when they are quite long
- Zooming in on a region, e.g. by setting loop start and end markers
- Starting playback from a certain position which is not the beginning

It's probably a pretty major undertaking but may be very useful.

**Comment by nickjillings** [comment]
I think we need to carefully plan exactly what the scrubber bar should be and its behaviour. It's probably one of the oldest parts of code that has yet to be refactored. Getting this right could be very cool. Plus, UI/UX design here!

**Comment by BrechtDeMan** [comment]
Agreed! Happy to be part of planning this (probably not implementing)

**Comment by nickjillings** [comment]
@djmoffat would be good for you to have some feedback here as well.

Ultimately it can do all of it, mostly it was implemented quickly and just left alone. It's purely writing some clever JS magic to handle clicks, but that's actually just small stuff mostly.

**Comment by djmoffat** [comment]
I'm not aware of any real requirement for that feature, as most of the time I use it is with fairly short samples of audio. Perhaps this ties in with the ability to annotate direct onto the waveform?

**Comment by BrechtDeMan** [comment]
There are a few studies on optimal length and many warn against samples which are too short. I'm personally involved in several which have 30s< samples (may be ill-advised). But it's something many participants have commented on, i.e. 'it would be nice if I could just go back a little bit instead of waiting for the whole thing to play again'. 

Of course this needs to be optional as sometimes you want to make sure participants don't focus on a particular glitch and instead consider the whole stimulus...

**Comment by djmoffat** [comment]
TL;DR yes, let's do it.

So, I agree that stimuli can be too short in the context of sound exerts, but not sure how to make a swoosh should longer...

I think it is important for people to be able to select where in the song to audition, so that they can choose to listen to a part of a mix they have not yet.

Regardless of the specific applications, if it is easy to implement, then it gives the test examiner more options for how to run the test, which I believe is fundamentally a good thing, so long as it is well structured and makes sense. 



---
Dave Moffat
Phone: +44 7969 375 889
Email: me@davemoffat.com
